src/vivarium/framework/randomness/index_map.py

```diff
@@ -49,10 +52,13 @@ def update(self, new_keys: pd.DataFrame) -> None:
         if len(final_keys) != len(final_keys.unique()):
             raise RandomnessError("Non-unique keys in index")
 
-        final_mapping = self._build_final_mapping(new_mapping_index)
+        final_mapping = self._build_final_mapping(new_mapping_index, clock_time)
 
         # Tack on the simulant index to the front of the map.
-        final_mapping.index = final_mapping_index
```

Comment:
This is the actual bugfix.

Reply:
So adding the clock should make it less likely to hit this edge case you saw?

Reply:
Correct, though that's not what this is doing. This is joining the simulant index back onto the map rather than brute-force setting the index